### PR TITLE
style: Assert layout data invariants are sane when querying layout.

### PR DESCRIPTION
**Pull request by emilio**
Trying to find a better test case for https://github.com/servo/servo/issues/14231.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14234)

### CODE REVIEW

File: components/layout/query.rs [hunk 1]
```diff
@@ -613,6 +613,15 @@ fn ensure_node_data_initialized<N: LayoutNode>(node: &N) {
...
         if current.borrow_layout_data().is_some() {
+            if cfg!(debug_assertions) {
+                cur = current.parent_node();
+                while let Some(current) = cur {
+                    assert!(current.borrow_layout_data().is_some(),
+                            "Broken invariant: Child has data without parent \
+                             having it");
+                    cur = current.parent_node();
+                }
+            }
```
Comment (nox) [inline components/layout/query.rs]: That looks pretty bad, given it's already in a loop that calls `parent_node`. As for the invariant, I'm not sure I should be the one to review this.

Comment (emilio) [inline components/layout/query.rs]: Yeah, sorry, this was only to pass it through automation and get some test results, should've closed this.

**Pull request closed without merging**